spec/price_calculator_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'price_calculator'
+require 'dish'
+
+describe PriceCalculator do
+  subject(:price_calculator) {described_class.new(selected_dishes)}
+  let(:selected_dishes) {double :selected_dishes}
+  let(:dish) {double :dish}
+  let(:dish2) {double :dish}
+  let(:dish3) {double :dish}
+  before do
+    allow(dish).to receive(:name) {:chow_mein}
+    allow(dish2).to receive(:name) {:special_fried_rice}
+    allow(dish3).to receive(:name) {:sweet_and_sour_chicken}
+    allow(dish).to receive(:price) {3.99}
+    allow(dish2).to receive(:price) {4.50}
+    allow(dish3).to receive(:price) {4.25}
+    allow(selected_dishes).to receive(:quantity) do
+      {dish => 2, dish2 => 4, dish3 => 1}
+    end
+    allow(selected_dishes).to receive(:each) do
+      selected_dishes.quantity.each do |dish,quantity|
+        price_calculator.grand_total += dish.price * quantity
+      end
+    end
+  end
+  xcontext '#initialize' do
+    it 'receives the selected_dishes from the menu' do
+      expect(price_calculator.selected_dishes).not_to be_empty
+    end
+  end
+
+  context '#price calculation' do
+    it 'calculates the price correctly' do
+      price_calculator.price(selected_dishes)
+      expect(price_calculator.grand_total).to eq(selected_dishes.quantity[dish] * dish.price + selected_dishes.quantity[dish2] * dish2.price + selected_dishes.quantity[dish3] * dish3.price)
```

Comment:
Metrics/LineLength: Line is too long. [189/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)